platform/FileHandle.h

```diff
@@ -152,8 +162,106 @@ class FileHandle {
      */
     MBED_DEPRECATED_SINCE("mbed-os-5.4", "Replaced by FileHandle::size")
     virtual off_t flen() { return size(); }
+
+    /** Set blocking or non-blocking mode of the file operation like read/write.
+     *  Definition depends upon the subclass implementing FileHandle.
+     *  The default is blocking.
+     *
+     *  @param blocking true for blocking mode, false for non-blocking mode.
+     */
+    virtual int set_blocking(bool blocking) {
+        return -1;
+    }
+
+    /** Check for poll event flags
+     * The input parameter can be used or ignored - the could always return all events,
+     * or could check just the events listed in events.
+     * Call is non-blocking - returns instantaneous state of events.
+     * Whenever an event occurs, the derived class must call _poll_change().
+     * @param events bitmask of poll events we're interested in - POLLIN/POLLOUT etc.
+     * @return bitmask of poll events that have occurred.
+     */
+    virtual short poll(short events) const {
```

Comment:
Ah fair point. Thoughts on adding a `poll_t`? (or `poll_mask_t` or something else).

In regards to overloading, `mbed::poll` is still able to overload poll through the confusing weirdness that is [ADL](http://en.cppreference.com/w/cpp/language/adl). It sounds like we should go with `poll` over `mbed_poll`.

Hmm, so what do you think we should do if a network stack can't provide a definite poll? It could just not allow non-block operation at that point. Is there some way for poll to indicate it's uncertain?